onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -312,27 +312,34 @@ def initializers(self) -> Mapping[str, torch.Tensor]:
 
     @beartype
     def add_input(
-        self, input_name: str, input_value: Optional[torch.Tensor] = None
+        self,
+        input_name: str,
+        input_value: Optional[torch.Tensor] = None,
+        dynamic_axes: bool = False,
     ) -> TorchScriptTensor:
         # TODO: Take in a TorchScriptTensor?
-        # TODO: Support dynamic shapes
         if input_value is None:
             # This input argument is None, which is mapped
             # to a NULL value in TorchScript type system.
             torch_value = _create_op_call_in_torch_graph(
                 self._torch_graph, "prim::Constant", inputs=(), attributes={}
             )[0]
             torch_value.setType(
-                torch._C.OptionalType.ofTensor()  # pylint: disable=c-extension-no-member,protected-access
+                torch.OptionalType.ofTensor()  # pylint: disable=c-extension-no-member,protected-access
             )
             tensor_value = _wrap_torch_value_to_tensor(torch_value)
             return tensor_value  # type: ignore[return-value]
         torch_value = self._torch_graph.addInput(input_name)
-        torch_value.setType(
-            torch._C.TensorType.create_from_tensor(  # pylint: disable=c-extension-no-member,protected-access
-                input_value
-            )
+        torchscript_input_value = torch.TensorType.create_from_tensor(  # pylint: disable=c-extension-no-member,protected-access
+            input_value
         )
+        if dynamic_axes:
+            rank = torchscript_input_value.dim()
+            # We set input sizes all to be dynamic to initialize the model conversion,
+            # and nodes are also set to be dynamic in exporter function: _fill_tensor_meta.
+            torch_value.setType(torchscript_input_value.with_sizes([None] * rank))
```

Comment:
I think we should grab the sizes from fx graph input? I'm ok with this for now since we have not hooked up dynamic export yet, but once that happens, we should just need to copypaste shape info from there. Do you see any concerns/issues? @titaiwangms @justinchuby

Reply:
For current solution (faster), for each input FakeTensor, we keep static part of shapes, and replace the SymInts with None, and still use `with_sizes()` to set it on TorchScriptGraph. The downside I guess it's losing information that "same SymInts indicates same shape", for example, [symint0, symint0, symint1] would all be [None, None, None]